### PR TITLE
Update for CASSANDRA-15299 protocol v5 changes

### PR DESCRIPTION
**Pull request by emerkle826**
This addresses the Cassandra 4.0 changes made as part of [CASSANDRA-15299](https://issues.apache.org/jira/browse/CASSANDRA-15299). Specifically, it changes the Netty channel pipeline initialization in the 4.x shim so that it works similar to how the Server changed for CASSANDRA-15299.

### CODE REVIEW

File: Dockerfile-oss
Comment (JeremiahDJordan) [inline Dockerfile-oss:58]: should we just only copy the jar that matches the C* version in the image?

File: management-api-server/pom.xml
Comment (JeremiahDJordan) [inline management-api-server/pom.xml:186]: Does this need a 4x version?

**Review by JeremiahDJordan — APPROVED** [review]
This looks good to me.  We should run this with cass-operator to double check everything there, since that is the main consumer, before merging.

File: management-api-server/pom.xml
Comment (emerkle826) [reply management-api-server/pom.xml:186]: As it stands now, it does not. This bit configures any unit tests that are run to have JVM_EXTRA_OPTS set, but none of the unit tests that run currently need this to be set to the 4.x agent. Most of the tests that are run are technically integration tests (though they run in the maven unit test phase), and spin up Docker containers with the appropriate image for the Cassandra/DSE version they are testing. The Docker images will only have the appropriate version of the agent jar in them in those tests.

**Review by tjake — CHANGES_REQUESTED** [review]
Thanks for taking this on.

Overall, makes sense but there is a lot of redundant code now.  

I would suggest making an agent-common package with all the shared classes.  You could use the existing common module even though the server doesn't need it.

* There are also classes that aren't used like 4x in the 3x agent.  
* I don't think we need the shim loaders anymore.

File: Dockerfile-4_0
Comment (emerkle826) [inline Dockerfile-4_0:20]: This needs to get added back in.

File: Dockerfile-astra-4_0
Comment (emerkle826) [inline Dockerfile-astra-4_0:6]: this needs to be updated like the 4.0 dockerfile

**Comment by emerkle826** [comment]
@tjake I've refactored the agents so there are no more _shim_ projects. In doing so, I've introduced a couple of service loaders so we don't have to use reflection/class names directly in the code, each agent will load the appropriate providers for the bits that need to be different/compiled against different Cassandra libraries. If you have a few minutes for another review, I'd appreciate it.

**Comment by JeremiahDJordan** [comment]
The refactoring looks good to me.  Thanks for the README about why some classes are duplicated.

**Review by tjake — APPROVED** [review]
LGTM thanks for cleaning that up!

**Pull request merged**
